feat: Spacecat task processor lambda

tkotthakota-adobe opened the pull request:
Create Spacecat task processor lambda as SQS consumer to receive messages (type, payload) via defined SQS queue. Based on the type, right task handler is invoked to fulfill the task. Slack context can be passed in the message to get the task response after completion. 

https://jira.corp.adobe.com/browse/SITES-32832
https://wiki.corp.adobe.com/display/AEMSites/Spacecat+Onboard+Automation

## Test result on slack channel:
<img width="1338" alt="image" src="https://github.com/user-attachments/assets/d96796ec-f103-4ad7-8a4d-f433f378e20d" />

lenaorobei commented on .eslintrc.cjs:
```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
```
Please check the other files too
```suggestion
 * Copyright 2025 Adobe. All rights reserved.
```

lenaorobei commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,74 @@
+# Contributing to Project Franklin
```
Probably, needs to be changed

lenaorobei commented on src/tasks/opportunity-status-processor/handler.js:
```diff
@@ -0,0 +1,108 @@
+    auditTypes,
+  });
+
+  await say(env, log, slackContext, ':information_source: *Checking opportunity status*');
```
```suggestion
```

lenaorobei commented on src/tasks/disable-import-audit-processor/handler.js:
```diff
@@ -0,0 +1,76 @@
+    importTypes,
+    auditTypes,
+  });
+  await say(env, log, slackContext, ':broom: *Disabling imports and audits*');
```
```suggestion
```

lenaorobei commented on src/tasks/disable-import-audit-processor/handler.js:
```diff
@@ -0,0 +1,76 @@
+    await configuration.save();
+    log.info('Database changes saved successfully');
+
+    const slackMessage = `:white_check_mark: Disabled imports ${JSON.stringify(importTypes)} and audits ${JSON.stringify(auditTypes)} for site ${siteId}`;
```
```suggestion
    const slackMessage = `:broom: Disabled imports: ${importTypes.join(', ')} and audits: ${auditTypes.join(', ')}`;
```

lenaorobei reviewed (commented):
Please see some minor comments

tkotthakota-adobe commented on src/support/sqs.js:
```diff
@@ -0,0 +1,57 @@
+/*
```
removed this file as this service only SQS consumer for now.

rpapani commented on scripts/watch-and-copy.js:
```diff
@@ -0,0 +1,82 @@
+const __dirname = path.dirname(__filename);
+
+const srcDir = path.join(__dirname, '..', 'src');
+const destDir = path.join(__dirname, '..', '.aws-sam/build/SpacecatAuditWorkerFunction/src');
```
looks like a copy over from audit

tkotthakota-adobe replied:
thanks for catching it

rpapani commented on src/tasks/demo-url-processor/handler.js:
```diff
@@ -0,0 +1,45 @@
+const TASK_TYPE = 'demo-url-processor';
+
+/**
+ * Runs the audit status processor
```
demo url processor

rpapani commented on src/tasks/disable-import-audit-processor/handler.js:
```diff
@@ -0,0 +1,68 @@
+    }
+    const siteConfig = site.getConfig();
+    for (const importType of importTypes) {
+      siteConfig.disableImport(importType);
```
can you check if its enabled, then only disable

tkotthakota-adobe replied:
What is advantage of checking if its enabled? Is disableImport expensive operation? or trying disableImport on already disabled one causes any issue? What is the api to check if import is enabled or where I can find it?

tkotthakota-adobe replied:
If we try to disable the import which already disabled, does not cause any issue? or performance reason?
Let me check what api I can use to check if it is already enabled.

tkotthakota-adobe replied:
@rpapani will address this part of this ticket: https://jira.corp.adobe.com/browse/SITES-33307

rpapani commented on src/tasks/opportunity-status-processor/handler.js:
```diff
@@ -0,0 +1,110 @@
+
+  try {
+    // Get the site and its opportunities
+    const site = await Site.findByBaseURL(`https://${siteId}.com`);
```
is siteId spacecat siteId or baseUrl? Please use the variable names accordingly. If you already have the siteId, please use `Site.findById` instead of `Site. findByBaseURL`

tkotthakota-adobe replied:
good point. It is siteId.

rpapani commented on src/tasks/opportunity-status-processor/handler.js:
```diff
@@ -0,0 +1,110 @@
+        // eslint-disable-next-line no-continue
+        continue;
+      }
+      processedTypes.add(opportunityType);
```
make sure you add the opportunity that has the suggestions, that way we know that there's an opportunity with suggestions

tkotthakota-adobe replied:
This step is merely to dedup opportunity types. later we are actually checking to see what opportunities have suggestions to create status accordingly.

rpapani reviewed (approved):
please incorporate the provided feedback, rest looks good 👍

tkotthakota-adobe commented on README.md:
````diff
@@ -0,0 +1,106 @@
+1. Clone the repository
+2. Install dependencies:
+   ```sh
+   npm ci
````
agreed.

tkotthakota-adobe commented on src/index.js:
```diff
@@ -0,0 +1,80 @@
+
+// Custom secret name resolver to use the correct secret path
+function getSecretName() {
+  return '/helix-deploy/spacecat-services/api-service/latest';
```
I agree if we need it we can create own store for this service.

tkotthakota-adobe commented on src/tasks/disable-import-audit-processor/handler.js:
```diff
@@ -0,0 +1,68 @@
+
+    const configuration = await Configuration.findLatest();
+    for (const auditType of auditTypes) {
+      configuration.disableHandlerForSite(auditType, site);
```
What is the api to check audits are already enabled or where I can find it?

tkotthakota-adobe replied:
@rpapani will address this part of this ticket: https://jira.corp.adobe.com/browse/SITES-33307

tkotthakota-adobe commented on src/tasks/demo-url-processor/handler.js:
```diff
@@ -0,0 +1,45 @@
+    organizationId,
+  });
+
+  const demoUrl = `${siteUrl}?organizationId=${organizationId}#/@aemrefdemoshared/sites-optimizer/sites/${siteId}/home`;
```
Thanks for clarifying about hard coding. 
Where can we find mapping for all ims orgs: imsOrg -> tenantId ?

tkotthakota-adobe replied:
Fixed as per the recommendation @rpapani

tkotthakota-adobe commented:
@rpapani addressed review comments. 
1. Under the secret name: /helix-deploy/spacecat-services/task-manager/latest below keys are added on dev. Same need to be added in prod:
IMS_ORG_TENANT_ID_MAPPINGS
SLACK_BOT_TOKEN
SLACK_SIGNING_SECRET
SLACK_TOKEN_WORKSPACE_INTERNAL
SLACK_OPS_CHANNEL_WORKSPACE_INTERNAL
SLACK_TOKEN_ADOBE_INTERNAL

2. We also need to add lambda trigger on prod env.

tkotthakota-adobe commented on package.json:
```diff
@@ -0,0 +1,132 @@
+  },
+  "devDependencies": {
+    "@adobe/eslint-config-helix": "2.0.9",
+    "@adobe/helix-deploy": "https://gitpkg.now.sh/alinarublea/helix-deploy?main",
```
Same is used in audit-worker. We will check on this later.
